Jnf/rating proposals

jnf opened the pull request:
Please don't judge too harshly. :(

jnf committed:
um, I didn't put the print files in the right place. This time maybe?

jnf committed:
another attempt to get the print stylesheet compiled.

jnf committed:
add some files to gitignore

jnf committed:
add a migration and model for proposal ratings

jnf committed:
Add a ratings controller with some view to support it.

jnf committed:
enable access to the ratings controller for admin and review users.

jnf committed:
some basic styles for the ratings section.

jnf committed:
adds a post route for ratings

jnf committed:
Look, I'm not proud of this code.

It's, like, O(n^infinity). We're not using this version of blinder ever again. We've got to adapt the data model to make it reliable, secure, and easy to retrieve specific elements from the submitted proposals. Like the friggin' title of the proposal. Ugh.

Anyway, this will get us over the hump maybe.

khjrtbrg commented on app/views/review/list.html.haml:
```diff
@@ -3,21 +3,19 @@
           %td
-            = link_to proposal.slug[0,6], review_path(@event, proposal), class: 'slug'
+            = link_to proposal.get_title, review_path(@event, proposal), class: 'slug'
+          %td.rating= "#{rating ? rating.score : 0}/5"
```
Could we not default it to 0 to make it easier to spot the ones you haven't rated yet (just in case someone wants to give an real rating of 0)?

khjrtbrg commented on app/views/shared/_rating.html.haml:
```diff
@@ -0,0 +1,9 @@
+  = form_tag '/rate' do
+    = hidden_field_tag :slug, proposal.event.slug
+    = hidden_field_tag :proposal_id, proposal.id
+    = hidden_field_tag :score, score
```
Could we make this a drop down so we can show them their saved score? Or indicate which one they selected somewhere in this partial for later reference?

jnf committed:
Manually assign score once a rating is found or initialized.

This fixes the bug where re-rating a proposal resulted in a new rating being created
instead of the existing rating being updates.

jnf committed:
Apply a visual indicator to a rating if the user has rated the proposal in the past.

There's an accessbility concern to address; I'll be working on that later.

jnf committed:
Use a hyphen instead of a zero to ident unreviewed proposals

jnf committed:
lock the rails version in the 4.2.* family; this formalizes the need for a patch route in the proposal safety review.

khjrtbrg commented:
lgtm :shipit:
